walk/walk.go

```diff
@@ -210,6 +210,58 @@ func Walk2(c *config.Config, cexts []config.Configurer, dirs []string, mode Mode
 		return errors.Join(w.errs...)
 	}
 
+	// Visit additional directories that extensions requested for indexing.
+	for len(w.relsToVisit) > 0 {
+		// Don't simply range over relsToVisit. We may append more.
+		relToVisit := w.relsToVisit[0]
+		w.relsToVisit = w.relsToVisit[1:]
+
+		dirToVisit := filepath.Join(c.RepoRoot, relToVisit)
+		if fi, err := os.Stat(dirToVisit); err != nil || !fi.IsDir() {
```

Comment:
Similar to a comment above.. what if `a/b/c` does not exist but `a/b` does? Can't we avoid this stat and just `break` the `for` below once we reach a non-existing directory?

Reply:
Yes, converted this to a check on whether `loadDirInfo` returns `fs.ErrNotExist`. So there's no extra I/O, and the behavior is probably a bit simpler to explain.